server/controllers/personalController.ts

```diff
@@ -175,4 +180,122 @@ export default class PersonalController {
       },
     }
   }
+
+  weight() {
+    return {
+      metric: {
+        edit: async (req: Request, res: Response, next: NextFunction) => {
+          const { prisonerNumber } = req.params
+          const { clientToken } = req.middleware
+          const prisonerData: Prisoner = req.middleware?.prisonerData
+          const fieldValueFlash = req.flash('fieldValue')
+          const errors = req.flash('errors')
+          const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+
+          res.render('pages/edit/weightMetric', {
+            pageTitle: 'Edit weight',
+            prisonerNumber,
+            breadcrumbPrisonerName: formatName(prisonerData.firstName, '', prisonerData.lastName, {
+              style: NameFormatStyle.lastCommaFirst,
+            }),
+            errors: hasLength(errors) ? errors : [],
+            fieldName: 'weight',
+            fieldValue: fieldValueFlash.length > 0 ? fieldValueFlash[0] : prisonPerson?.physicalAttributes.weight,
+          })
+        },
+
+        submit: async (req: Request, res: Response, next: NextFunction) => {
+          const { prisonerNumber } = req.params
+          const { clientToken } = req.middleware
+          const { editField } = req.body
+          const errors: HmppsError[] = []
+          const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+
+          const weight = parseInt(editField, 10)
+          if (Number.isNaN(weight) || weight <= 0) {
+            errors.push({ text: 'Enter a number greater than 0' })
+          }
```

Comment:
Didn't we decide not to instantiate and mutate variable?
I would recommend using this as a guard and returning from here then you can remove the if below.

Something like
```javascript
if (Number.isNaN(weight) || weight <= 0) {
     req.flash('errors', [{ text: 'Enter a number greater than 0' }])
     return res.redirect(`/prisoner/${prisonerNumber}/personal/edit/weight`)
 }

try {
  ...
```